component/stream/read_write_filename.go

```diff
@@ -0,0 +1,478 @@
+		if err != nil && err != io.EOF {
+			log.Err("Stream::WriteFile : error failed to write data to %s: [%s]", options.Handle.Path, err.Error())
+		}
+		return data, err
+	}
+	written, err := rw.readWriteBlocks(options.Handle, options.Offset, options.Data, true)
+	if err != nil {
+		log.Err("Stream::WriteFile : error failed to write data to %s: [%s]", options.Handle.Path, err.Error())
+	}
+	options.Handle.Flags.Set(handlemap.HandleFlagDirty)
+	return written, err
+}
+
+// TODO: truncate in cache
+func (rw *ReadWriteFilenameCache) TruncateFile(options internal.TruncateFileOptions) error {
+	// log.Trace("Stream::TruncateFile : name=%s, size=%d", options.Name, options.Size)
+	if !rw.StreamOnly {
+		rw.purge(options.Name, false)
+	}
+	err := rw.NextComponent().TruncateFile(options)
+	if err != nil {
+		log.Err("Stream::TruncateFile : error truncating file %s [%s]", options.Name, err.Error())
+		return err
+	}
+	return nil
+}
+
+func (rw *ReadWriteFilenameCache) RenameFile(options internal.RenameFileOptions) error {
+	// log.Trace("Stream::RenameFile : name=%s", options.Src)
+	if !rw.StreamOnly {
+		rw.purge(options.Src, false)
+	}
+	err := rw.NextComponent().RenameFile(options)
+	if err != nil {
+		log.Err("Stream::RenameFile : error renaming file %s [%s]", options.Src, err.Error())
+		return err
+	}
+	return nil
+}
+
+func (rw *ReadWriteFilenameCache) CloseFile(options internal.CloseFileOptions) error {
+	// log.Trace("Stream::CloseFile : name=%s, handle=%d", options.Handle.Path, options.Handle.ID)
+	// try to flush again to make sure it's cleaned up
+	err := rw.FlushFile(internal.FlushFileOptions(options))
+	if err != nil {
+		log.Err("Stream::FlushFile : error flushing file %s [%s]", options.Handle.Path, err.Error())
+		return err
+	}
+	if !rw.StreamOnly {
+		rw.purge(options.Handle.Path, true)
+	}
+	err = rw.NextComponent().CloseFile(options)
+	if err != nil {
+		log.Err("Stream::CloseFile : error closing file %s [%s]", options.Handle.Path, err.Error())
+	}
+	return err
+}
+
+func (rw *ReadWriteFilenameCache) FlushFile(options internal.FlushFileOptions) error {
+	// log.Trace("Stream::CloseFile : name=%s, handle=%d", options.Handle.Path, options.Handle.ID)
+	err := rw.NextComponent().FlushFile(options)
+	if err != nil {
+		log.Err("Stream::FlushFile : error flushing file %s [%s]", options.Handle.Path, err.Error())
+		return err
+	}
+	options.Handle.Flags.Clear(handlemap.HandleFlagDirty)
+	return nil
+}
+
+// TODO: any cached file to be purged without flushing
+func (rw *ReadWriteFilenameCache) DeleteFile(options internal.DeleteFileOptions) error {
+	// log.Trace("Stream::DeleteFile : name=%s", options.Name)
+	if !rw.StreamOnly {
+		rw.purge(options.Name, false)
+	}
+	err := rw.NextComponent().DeleteFile(options)
+	if err != nil {
+		log.Err("Stream::DeleteFile : error deleting file %s [%s]", options.Name, err.Error())
+		return err
+	}
+	return nil
+}
+
+// TODO: any cached file to be purged without flushing
```

Comment:
Are these todos still valid, or they have been already taken care of

Reply:
I have added all those changes will remove the todos